sdk/core/azure-core/inc/azure/core/internal/tracing/tracing_impl.hpp

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Handling log messages from Azure SDK.
+ */
+
+#pragma once
+
+#include "azure/core/datetime.hpp"
+#include "azure/core/internal/extendable_enumeration.hpp"
+#include "azure/core/nullable.hpp"
+#include "azure/core/tracing/tracing.hpp"
+#include "azure/core/url.hpp"
+#include <array>
+#include <memory>
+#include <string>
+#include <vector>
+
+// Forward declare Azure::Core::Http::Request to resolve an include file ordering problem.
+namespace Azure { namespace Core { namespace Http {
+  class Request;
+}}} // namespace Azure::Core::Http
```

Comment:
Unrelated to this PR:
This sort of approach adds a bit of fragility in how our headers depend on each other, and might indicate a layering issue in the API.

We might want to re-visit why this was needed, and if there's an alternative approach:
> virtual void PropagateToHttpHeaders(Azure::Core::Http::Request& request) override;

Reply:
IMHO, this change was explicitly made to *remove* fragility in our header ordering and to fix a circular dependency issue.

`PropagateToHttpHeaders` was needed because there are OpenTelemetry headers which need to be injected into each HTTP request. 

Without significant refactoring of the `Request` object to change how headers are managed, this was the simplest change to resolve the issue.

The *implementation* of `PropagateToHttpHeaders` needs to know the definition of `Request`. The declaration of `PropagateToHttpHeaders` does not need to know the definition.